packages/language-server/src/server.ts

```diff
@@ -145,47 +137,28 @@ export function startServer(options?: LSPOptions): void {
     documentSettings.delete(e.document.uri)
   })
 
-  function getDocumentSettings(resource: string): Thenable<LSPSettings> {
-    if (!hasConfigurationCapability) {
-      connection.console.info(`Using default prisma-fmt binary path.`)
-      return Promise.resolve(globalSettings)
-    }
-    let result = documentSettings.get(resource)
-    if (!result) {
-      result = connection.workspace.getConfiguration({
-        scopeUri: resource,
-        section: 'prisma',
-      })
-      documentSettings.set(resource, result)
-    }
-    return result
-  }
-
-  function getPrismaFmtBinPath(binPathSetting: string | undefined): string {
-    if (!binPathSetting || binPathSetting.length === 0) {
-      return defaultBinPath
-    } else if (!existsSync(binPathSetting)) {
-      connection.window.showErrorMessage(
-        `Path to prisma-fmt binary (${binPathSetting}) does not exist. Using default prisma-fmt binary path instead.`,
-      )
-      return defaultBinPath
-    } else {
-      connection.console.log(
-        'Using binary path from Prisma Language Server configuration.',
-      )
-      return binPathSetting
-    }
-  }
+  // function getDocumentSettings(resource: string): Thenable<LSPSettings> {
+  //   if (!hasConfigurationCapability) {
+  //     connection.console.info(`Using default prisma-fmt binary path.`)
+  //     return Promise.resolve(globalSettings)
+  //   }
+  //   let result = documentSettings.get(resource)
+  //   if (!result) {
+  //     result = connection.workspace.getConfiguration({
+  //       scopeUri: resource,
+  //       section: 'prisma',
+  //     })
+  //     documentSettings.set(resource, result)
+  //   }
+  //   return result
+  // }
 
   async function validateTextDocument(
     textDocument: TextDocument,
   ): Promise<void> {
-    const settings = await getDocumentSettings(textDocument.uri)
```

Comment:
Are we sure this is a good idea to remove all of this? Seems like a general settings handling for me - currently just only used for the binary path but potentially also useful for other things. We should probably pick that apart and re-add it @tomhoule

Reply:
I wouldn't have removed all that, but the linter complains about unused code, and I didn't have the energy to get involved into tsconfig/eslint config.